lib/ui/painting.dart

```diff
@@ -23,13 +23,16 @@ part of dart.ui;
 
 bool _rectIsValid(Rect rect) {
   assert(rect != null, 'Rect argument was null.');
-  assert(!rect._value.any((double value) => value.isNaN), 'Rect argument contained a NaN value.');
+  assert(!(rect.left.isNaN || rect.top.isNaN || rect.right.isNaN || rect.bottom.isNaN), 'Rect argument contained a NaN value.');
```

Comment:
if we're saying these aren't valid, should we assert they can't be created?

Reply:
i guess we're just saying that these can't be used in the canvas apis, not that they're invalid in general.

maybe have an hasNaN getter on Rect and RRect?

Reply:
As public or private API?

Reply:
public would be fine.

Reply:
Done.